RELEASE/scripts/autoscend.ash

```diff
@@ -2384,20 +2384,140 @@ boolean powerLevelAdjustment()
 	return false;
 }
 
+boolean unlockThinknerdWarehouse()
+{
+	//returns true if you have access to [The Thinknerd Warehouse], if you don't have access tries to unlock it and then return true.
+	//returns false if fails to unlock.
+	//much easier to do if you already have torso awaregness
+	
+	if(internalQuestStatus("questM22Shirt") > -1)
+	{
+		return true;
+	}
+	
+	//unlocking is a multi step process. We want to try things in reverse to conserve resources and in case some steps were already complete.
+	
+	boolean useLetter()
+	{
+		if(item_amount($item[Letter for Melvign the Gnome]) > 0)
+		{
+			if(use(1, $item[Letter for Melvign the Gnome]))
+			{
+				return true;
+			}
+			else
+			{
+				abort("Somehow failed to use [Letter for Melvign the Gnome]... aborting to prevent infinite loops");
+			}
+		}
+		return false;
+	}
+	
+	item target_shirt = $item[none];		//needs to be global so we can call it outside boolean hasShirt()
+	boolean hasShirt()
+	{
+		if(target_shirt != $item[none]) return true;
+		int[item] inventory_snapshot = get_inventory();		//need to refresh inventory_snapshot every time this function is called.
+		foreach it in inventory_snapshot
+		{
+			if(item_type(it) == "shirt" && can_equip(it))
+			{
+				target_shirt = it;
+				break;
+			}
+		}
+		if(target_shirt != $item[none]) return true;
+		return false;
+	}
+	boolean useShirtThenLetter()
+	{
+		equip($slot[shirt], target_shirt);
+		if(useLetter()) return true;
+		auto_log_error("For some reason boolean unlockThinknerdWarehouse() failed when trying to use the shirt [" + target_shirt + "] to get [Letter for Melvign the Gnome] to start the quest", "red");
+		return false;
+	}
+	void tryShirtCreate(item it)
+	{
+		if(!hasShirt() && can_equip(it) && creatable_amount(it) > 0)
+		{
+			if(create(1, it))
+			{
+				target_shirt = it;
+			}
+		}
+	}
+	void tryShirtPull(item it)
+	{
+		if(!hasShirt() && can_equip(it) && canPull(it))
```

Comment:
I think `canPull(it)` is redundant here

Reply:
Can pull checks:
1. is it in standard
2. are we in hardcore
3. do we have pulls left

can equip checks if we have enough mus, mox, or mys to equip it.

Reply:
Oh... actually can_equip also checks if you have torso awaregness. So it needs to be removed.
equip() also checks if you have it too so i need to handle this via visit_url too...